packages/explorer/src/observer/decorator.ts

```diff
@@ -19,52 +21,100 @@ export function observer({ explorerUrl = "http://localhost:13690", waitForTransa
   account extends Account | undefined = Account | undefined,
 >(
   client: Client<transport, chain, account>,
-) => Pick<WalletActions<chain, account>, "writeContract"> {
+) => Pick<WalletActions<chain, account>, "writeContract"> & Pick<BundlerActions, "sendUserOperation"> {
   const emit = createBridge({ url: `${explorerUrl}/internal/observer-relay` });
 
-  return (client) => ({
-    async writeContract(args) {
-      const writeId = `${client.uid}-${++writeCounter}`;
-      const write = getAction(client, writeContract, "writeContract")(args);
-
-      // `writeContract` above will throw if this isn't present
-      const functionAbiItem = getAbiItem({
-        abi: args.abi,
-        name: args.functionName,
-        args: args.args,
-      } as never)!;
-
-      emit("write", {
-        writeId,
-        address: args.address,
-        from: client.account!.address,
-        functionSignature: formatAbiItem(functionAbiItem),
-        args: (args.args ?? []) as never,
-        value: args.value,
-      });
-      Promise.allSettled([write]).then(([result]) => {
-        emit("write:result", { ...result, writeId });
-      });
-
-      write.then((hash) => {
-        const receipt = getAction(client, waitForTransactionReceipt, "waitForTransactionReceipt")({ hash });
-        emit("waitForTransactionReceipt", { writeId, hash });
-        Promise.allSettled([receipt]).then(([result]) => {
-          emit("waitForTransactionReceipt:result", { ...result, writeId });
+  return (client) => {
+    return {
```

Comment:
```suggestion
  return (client) => ({
```
small thing to help with the diff + reduce nesting